core/src/main/kotlin/com/gitlab/kordlib/core/Util.kt

```diff
@@ -48,11 +49,29 @@ internal inline fun <T> catchNotFound(block: () -> T): T?  {
     return try {
         block()
     } catch (exception: RestRequestException) {
-        if (exception.code == 404) null
+        if (exception.status.code == 404) null
         else throw exception
     }
 }
 
+@OptIn(ExperimentalContracts::class)
+internal inline fun <T> catchDiscordError(vararg codes: JsonErrorCode, block: () -> T): T?  {
+    contract {
+        callsInPlace(block, InvocationKind.EXACTLY_ONCE)
+    }
+    return try {
+        block()
+    } catch (exception: RestRequestException) {
+        when {
+            codes.isEmpty() -> null
```

Comment:
if codes is empty then we are not supposed to catch any json error.

Reply:
That would make the function effectively useless though. 
I rationalized that not providing any codes implied catching them all, but I see how that is counter-intuitive. Ideally I'd have the parameter default to all enums (since that is what it implied), but varargs cannot have default values.

Reply:
It's fine like this :+1: unless it's used we want to catch something we are not going to use this